src/libraries/beatmap/BeatmapLibrary.ts

```diff
@@ -64,4 +66,17 @@ export default class BeatmapLibrary {
   public static RemoveBeatmap(beatmap: BeatmapLocal) {
     store.commit("beatmap/removeBeatmap", { beatmap });
   }
+
+  public static GetPlaylists(beatmap: BeatmapLocal): PlaylistLocal[] {
+    const affiliatedPlaylists: PlaylistLocal[] = [];
+    PlaylistLibrary.GetAllValidPlaylists().forEach((playlist) => {
+      for (const map of playlist.maps) {
+        if (map.hash !== undefined && map.hash === beatmap.hash) {
+          affiliatedPlaylists.push(playlist);
+          break;
+        }
+      }
+    });
+    return affiliatedPlaylists;
```

Comment:
Can be simplified to
```ts
return PlaylistLibrary.GetAllValidPlaylists().filter((playlist) =>
  playlist.maps.find(
    (map) => map.hash === beatmap.hash && map.hash !== undefined
  )
);
```

Reply:
yeah, that's way better. For the sake of lazy evaluation I will put the undefined check up front though